app/src/lib/git/rebase.ts

```diff
@@ -0,0 +1,159 @@
+import * as Path from 'path'
+import * as FSE from 'fs-extra'
+
+import { Repository } from '../../models/repository'
+import { git } from './core'
+import {
+  WorkingDirectoryFileChange,
+  AppFileStatusKind,
+} from '../../models/status'
+import { ManualConflictResolution } from '../../models/manual-conflict-resolution'
+import { stageManualConflictResolution } from './stage'
+import { stageFiles } from './update-index'
+import { GitError, IGitResult } from 'dugite'
+import { getStatus } from './status'
+
+/**
+ * Check the `.git/REBASE_HEAD` file exists in a repository to confirm
+ * a rebase operation is underway.
+ */
+export function isRebaseHeadSet(repository: Repository) {
+  const path = Path.join(repository.path, '.git', 'REBASE_HEAD')
+  return FSE.pathExists(path)
+}
+
+/**
+ * A stub function to use for initiating rebase in the app.
+ *
+ * If the rebase fails, the repository will be in an indeterminate state where
+ * the rebase is stuck.
+ *
+ * If the rebase completes without error, `featureBranch` will be checked out
+ * and it will probably have a different commit history.
+ *
+ * @param baseBranch the ref to start the rebase from
+ * @param featureBranch the ref to rebase onto `baseBranch`
+ */
+export async function rebase(
+  repository: Repository,
+  baseBranch: string,
+  featureBranch: string
+) {
+  return await git(
+    ['rebase', baseBranch, featureBranch],
+    repository.path,
+    'abortRebase',
```

Comment:
Is this meant to be `abortRebase`? I kinda expected it would be `rebase`.

Reply:
Good spot! Fixed in 4249f7427